File: python/P165.py
class Solution(object):
    def compareVersion(self, version1, version2):
        version1_list = version1.split('.')
        version2_list = version2.split('.')
        overlap = max(len(version1_list), len(version2_list))
        for x in range(overlap):
            v1_tok = 0
            if x < len(version1_list):
                v1_tok = int(version1_list[x])
            v2_tok = 0
            if x < len(version2_list):
                v2_tok = int(version2_list[x])
            if v1_tok < v2_tok:
                return -1
            if v1_tok > v2_tok:
                return 1
        return 0

File: java/P165.java
class Solution {
    public int compareVersion(String version1, String version2) {
        String[] v1 = version1.split("\\.");
        String[] v2 = version2.split("\\.");
        // can use either larger length or shorter length to compare
        int length = Math.max(v1.length, v2.length);
        for (int i = 0; i < length; i++) {
            int l1 = i<v1.length? Integer.parseInt(v1[i]):0;
            int l2 = i<v2.length? Integer.parseInt(v2[i]):0;
            if (l1 > l2) return 1;
            else if (l1 < l2) return -1;
            else continue;
        }
        return 0;
    }
}